### PR TITLE
feat: Filtering UI down to Individual test blocks

### PR DESCRIPTION
**Pull request by g4rry420**
Resolves: #3109 

I have considered two ways to add this functionality:
- In first method, we will get index and nestedIndex of items when user clicks on test name and if the that indexes are present then we will show that code block. Potential Issue: If the test file is not formatted properly, i.e, starting and ending of test block on the same line, then it might not work
- Second Method (fallback) - Here, based on the test name, we will search for that test and filter the string block for that test. Potential issue with this fallback method is that if there are two tests with the same name, then it will show the test block whose name comes first in the string.

 There is also Codemirror add-ons for code folding which can be useful for this feature, but I haven't taken a deep look into it. https://codemirror.net/docs/ref/#language.foldCode

Video:
https://www.loom.com/share/9fdbf88ef5114b2fa76e45b119b31e61

Disclaimer: I am coming from React background, so I am learning Vue on the go, please let me know if I am missing best practices specific to Vue

Also, when I was trying to make commit, I am facing this eslint error, not sure why that occurred, so I have commited the files with `-n` flag.
Screenshot 2023-06-04 003953

### CODE REVIEW

**Comment by userquin** [comment]
run `nr lint` in your local, there are a few lint errors, check `Files changed` tab.

File: packages/ui/client/components/views/ViewEditor.vue [hunk 1]
```diff
@@ -120,13 +244,22 @@ async function onSave(content: string) {
...
-    v-model="code"
-    h-full
-    v-bind="{ lineNumbers: true }"
-    :mode="ext"
-    data-testid="code-mirror"
-    @save="onSave"
-  />
+  <div>
+    <IconButton
+      v-if="testIndex !== null || isDescribeBlock !== null || selectedTest !== null"
```
Comment (dammy001) [inline packages/ui/client/components/views/ViewEditor.vue:249]: Can we create a computed property that holds this to avoid duplicate.

Comment (g4rry420) [reply packages/ui/client/components/views/ViewEditor.vue:253]: Sure

**Comment by dammy001** [comment]
@userquin take a look at this again.

**Pull request closed without merging**